### DISCUSSION TITLE
new Model intances with new inertia form helper

### DISCUSSION THREAD

**Discussion by RUDIWER** [inertiajs/inertia#496] [Help]
Laravel  - Vue - Inertiajs new form helper

In my controller in make a new model instance for example :   $customer =  new Customer,
When i come in my vue component i inject thsi new model instance to the form object :   form: this.$inertia.form({...this.customer }),

This form right now has no fields in it.
I fill my form and add several fields to the form object.

When i post the form with :  this.form.post(this.route('customers.store', this.customer.id)
The request does NOT have the added fields ????

My soluton for the moment is to add all the fields in de Customer model with  :   protected $attributes = [ ...... ]

But this seems not a elegant solution for me, anyone a idea how to solve this a better way ?

**Comment by ajnsn** [comment]
@RUDIWER Currently the form helper only submits data of properties that were initialized when calling `this.$inertia.form()`. All other properties will be populated in the form e.g. via `v-model` but they will not be included in the submit data.

Behind the scenes the form helper needs this default properties to "know" which are relevant to the form and which are just internal  properties. 

When you create a new instance of your customer model via `new Customer()` the model is empty, e.g.

```php
$customer = new Customer();
$customer->toArray();
```

This will return an empty customer array. This explains why the form data is empty. There are no default properties.

Of course you can set the attributes if that works for you.
I would either assign all properties in Inertia's form instance:

```js
this.$inertia.form({
  name: this.customer.name,
  ...
}),

```

or when instanating the model in Laravel:

```php
$customer = new Customer([
  'name' => null,
]);
```

Hope this helps!

**Comment by RUDIWER** [comment]
Are there caveats on using  protected $attributes = [ ...... ] annd add all the fields there ?
That way on creating the new Customer allthe fields are in it.

**Reply by ajnsn** [reply]
@RUDIWER I do not think that there are any issues with that as long as you won't need any relationship etc. I personally do not like all this hidden Laravel magic stuff because it will impossible for another developer to *read* your code and see your intention *why* you did it. 

You could also take a look at [Laravel's API resouces](https://laravel.com/docs/8.x/eloquent-resources). As they are directly not meant for this, they work pretty well with Inertia and I often use them, e.g.

```php
return Inertia::render('component', [
    'customer' => function () {
        return CustomerResource::make(new Customer);
    },
]);
```

The cool thing is that you have a single set of properties and  property assignment. You can also use it with a Collection. Once you get your head around it, you'll love it. And it's easy to *read*.
